addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.konnected.internal.KonnectedConfiguration;
+import org.openhab.binding.konnected.internal.KonnectedHTTPUtils;
+import org.openhab.binding.konnected.internal.gson.KonnectedModuleGson;
+import org.openhab.binding.konnected.internal.gson.KonnectedModulePayload;
+import org.openhab.binding.konnected.internal.servelet.KonnectedHTTPServlet;
+import org.openhab.binding.konnected.internal.servelet.KonnectedWebHookFail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link KonnectedHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Zachary Christiansen - Initial contribution
+ */
+public class KonnectedHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(KonnectedHandler.class);
+    private KonnectedConfiguration config;
+    private KonnectedHTTPServlet webHookServlet;
+    private final KonnectedHTTPUtils http = new KonnectedHTTPUtils();
+    private String callbackIpAddress = null;
+    private String moduleIpAddress;
+    private Gson gson = new GsonBuilder().create();
+
+    /**
+     * This is the constructor of the Konnected Handler.
+     *
+     * @param thing          the instance of the Konnected thing
+     * @param webHookServlet the instance of the callback servlet that is running for communication with the Konnected
+     *                           Module
+     * @param hostAddress    the webaddress of the OpenHAB server instance obtained by the runtime
+     * @param port           the port on which the OpenHAB instance is running that was obtained by the runtime.
+     */
+    public KonnectedHandler(Thing thing, KonnectedHTTPServlet webHookServlet, String hostAddress, String port) {
+        super(thing);
+        this.webHookServlet = webHookServlet;
+        callbackIpAddress = hostAddress + ":" + port;
+        logger.debug("The callback ip address is: {}", callbackIpAddress);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // get the zone number in integer form
```

Comment:
you need to also process REFRESH commands - if you receive them for a channel, try and get a current state for it.

Reply:
added the RefreshType command and then added methods to process and send get request to module for state update on that channel